services/forms/repo_tag_form.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package forms
+
+import (
+	"net/http"
+
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/web/middleware"
+
+	"gitea.com/go-chi/binding"
+)
+
+// ProtectTagForm form for changing protected tag settings
+type ProtectTagForm struct {
+	NamePattern    string `binding:"Required;GlobPattern"`
```

Comment:
Should we have a length limit?

Reply:
https://github.com/go-gitea/gitea/blob/2c57352a30307ba274651ff2a93cc12b4afe9316/services/forms/repo_form.go#L630
The release page uses 255. A pattern may be longer but 255 should still be enough.